File: python/promptspec/models/llm_parameters.py
"""
Common LLM parameters that can be used across different models.
"""
from typing import Optional, List
from pydantic import BaseModel, Field, ConfigDict


class LLMParameters(BaseModel):
    """Common LLM parameters that can be used across different models."""
    
    model_config = ConfigDict(
        populate_by_name=True,
        alias_generator=lambda field_name: {
            'top_p': 'topP',
            'max_tokens': 'maxTokens',
            'stop_sequences': 'stopSequences'
        }.get(field_name, field_name)
    )
    
    temperature: Optional[float] = Field(
        None, 
        ge=0.0, 
        le=2.0, 
        description="Controls randomness of output (0-2.0)"
    )
    top_p: Optional[float] = Field(
        None, 
        ge=0.0, 
        le=1.0, 
        description="Nucleus sampling parameter (0-1.0)"
    )
    max_tokens: Optional[int] = Field(
        None, 
        gt=0, 
        description="Maximum number of tokens to generate"
    )
    stop_sequences: Optional[List[str]] = Field(
        None, 
        description="List of strings that stop generation when encountered"
    )

File: src/Models/LLMParameters.cs
using System.Collections.Generic;
using YamlDotNet.Serialization;

namespace PromptSpec.Models
{

    /// <summary>
    /// Represents common LLM parameters that can be used across different models.
    /// </summary>
    public class LLMParameters
    {
        /// <summary>
        /// Controls the randomness of the output. Value between 0 and 2.0.
        /// </summary>
        [YamlMember(Alias = "temperature")]
        public double? Temperature { get; set; }

        /// <summary>
        /// Nucleus sampling parameter. Value between 0 and 1.0.
        /// </summary>
        [YamlMember(Alias = "topP")]
        public double? TopP { get; set; }

        /// <summary>
        /// The maximum number of tokens to generate.
        /// </summary>
        [YamlMember(Alias = "maxTokens")]
        public int? MaxTokens { get; set; }

        /// <summary>
        /// A list of strings that will stop the generation when encountered.
        /// </summary>
        [YamlMember(Alias = "stopSequences")]
        public List<string> StopSequences { get; set; }
    }
}
